### PR TITLE
Added legend to zoomable pack diagram

### PR DESCRIPTION
**Pull request by aidan-epperly**
This PR adds a legend to the zoomable pack diagram in order to make the visualization more comprehensible.

### CODE REVIEW

**Comment by LRWeber** [comment]
Would it be possible to add some sort of background box to the legend? I think it would make it more legible once you are zoomed in and it starts to overlap parts of the graph.

**Comment by aidan-epperly** [comment]
Yes, it would. Do you want it to be fully opaque or somewhat transparent?

**Comment by LRWeber** [comment]
A slight transparency might be good! It makes it clearer how it's overlapping.

**Comment by aidan-epperly** [comment]
Here is the box. It doesn't automatically size to the text, but that may be unnecessary:
image

File: js/explore/pack_hierarchy.js [hunk 1]
```diff
@@ -210,11 +213,73 @@ function draw_pack_hierarchy(areaID) {
...
+            });
+
+            legend.append('rect')
+                .attr('fill', '#FFFFFF')
+                .attr('fill-opacity', 0.7)
+                .attr('height', labels.length * (legendRectSize + legendSpacing) + legendSpacing)
+                .attr('width', 160)
+                .attr('y', 0 - legendSpacing)
+                .attr('x', -5)
+                .attr('rx', 10);
```
Comment (LRWeber) [inline js/explore/pack_hierarchy.js:238]: The contents of the legend shouldn't be changing unless we change them, so I agree that a static box should be fine.

I have just a couple cosmetic change suggestions.
```suggestion
            legend.append('rect')
                .attr('fill', '#FFFFFF')
                .attr('fill-opacity', 0.9)
                .attr('height', labels.length * (legendRectSize + legendSpacing) + legendSpacing)
                .attr('width', 150)
                .attr('y', 0 - legendSpacing)
                .attr('x', -5)
                .attr('rx', 10);
```

**Comment by aidan-epperly** [comment]
image

**Pull request merged**